src/core/linter-rules/check-charset.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Linter rule "check-charset".
+ *
+ * Checks whether the document has `<meta charset="utf-8" />` properly.
+ */
+import LinterRule from "../LinterRule";
+import { lang as defaultLang } from "../l10n";
+
+const name = "check-charset";
+const meta = {
+  en: {
+    description: `Document must only contain one \`<meta>\` tag with charset set to 'utf-8'`,
+    howToFix: `Add this line in your document \`<head>\` section -  \`<meta charset="utf-8" />\` or set charset to "utf-8" if not set already.`,
+  },
+};
+
+// Fall back to english, if language is missing
+const lang = defaultLang in meta ? defaultLang : "en";
+const getCharset = element => {
+  return !/charset="(utf-8|UTF-8)"/.test(element);
+};
+
+/**
+ * Runs linter rule.
+ *
+ * @param {Object} config The ReSpec config.
+ * @param  {Document} doc The document to be checked.
+ */
+function linterFunction(conf, doc) {
+  const metas = doc.querySelectorAll("meta[charset]");
+  const val = [];
+  for (let i = 0; i < metas.length; i++) {
+    val.push(metas[i].outerHTML);
```

Comment:
Maybe we can push `meta.getAttribute("charset")` instead? Then we can skip `charset=` part in `getCharset` function.

Reply:
Also we prefer for-of loop: `for (const meta of metas)`

Reply:
Question: Why would there be multiple `meta[charset]`?

Reply:
@sidvishnoi No reason, it would always be by accident.

Reply:
@sidvishnoi just considering a hypothetical case, cases wherein if someone just copied html from some other source, yet very rare